styleguide/checkstyle.xml

```diff
@@ -139,7 +139,6 @@ module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
     <module name="OneStatementPerLine" />
     <module name="MultipleVariableDeclarations" />
     <module name="ArrayTypeStyle" />
-    <module name="MissingSwitchDefault" />
```

Comment:
So the rationale for this check is the following:
> It's usually a good idea to introduce a default case in every switch statement. Even if the developer is sure that all currently possible cases are covered, this should be expressed in the default branch, e.g. by using an assertion. This way the code is protected against later changes, e.g. introduction of new types in an enumeration type.

I agree it's a bit annoying in the case when you're using it in the same class where the enum is defined; suppression seems reasonable in that case.

Reply:
But checkstyle already warns on fallthrough. if you aren't using case fallthrough (which modern C++ compilers warn about, idk about javac) an empty default case is equivalent to adding an empty else clause on an if statement. That is, it's useless and doesn't protect against anything.

Reply:
Turns out the C++ compiler is smarter than the Java one; javac doesn't warn on missing cases. I added a throwing default case to all of them.